### PR TITLE
`Development`: Improve e2e cypress tests for quiz assessment in exams

### PR DESCRIPTION
**Pull request by rs-maker**
### Motivation and Context
 The exam assessment test is missing quiz assessment.

### Description
 I will cover the assessment of quiz exercises in this PR.

### Steps for Testing

I have created a tutorial for how to test cypress PRs <a href="https://confluence.ase.in.tum.de/display/ArTEMiS/Cypress+Tests">HERE</a> 
Please test all .spec files that were changed in this PR or ideally all of the existing ones.

### Review Progress

- Code Review
  - [x] Review 1
  - [ ] Review 2
- Manual Tests
  - [x] Test 1
  - [ ] Test 2

### CODE REVIEW

**Comment by rs-maker** [comment]
Tests passed a few times on test plan <a href="https://bamboo.ase.in.tum.de/browse/ARTEMIS-AETBBT-462"> here </a>

File: src/test/cypress/integration/exam/ExamAssessment.spec.ts [hunk 1]
```diff
@@ -124,11 +130,64 @@ describe('Exam assessment', () => {
...
+
+        beforeEach('Create exercise and submission', () => {
+            courseManagementRequests.createQuizExercise({ exerciseGroup }, [multipleChoiceQuizTemplate], 'Cypress Quiz').then(() => {
+                courseManagementRequests.generateMissingIndividualExams(exam);
+                courseManagementRequests.prepareExerciseStartForExam(exam);
+                cy.login(student, '/courses/' + course.id + '/exams/' + exam.id);
+                examStartEnd.startExam();
+                cy.contains('Cypress Quiz').click();
+                cy.get('#answer-option-0 > .selection > .ng-fa-icon > .svg-inline--fa').click();
+                cy.get('#answer-option-2 > .selection > .ng-fa-icon > .svg-inline--fa').click();
```
Comment (jhainzlschmid) [inline src/test/cypress/integration/exam/ExamAssessment.spec.ts:150]: This seems to be the same selector as in `MultipleChoiceQuiz#tickAnswerOption`, so you can probably use that page object here.

File: src/test/cypress/integration/exam/ExamAssessment.spec.ts [hunk 2]
```diff
@@ -124,11 +130,64 @@ describe('Exam assessment', () => {
...
+                cy.get('#exam-navigation-bar').find('.btn-danger').click();
```
Comment (jhainzlschmid) [inline src/test/cypress/integration/exam/ExamAssessment.spec.ts:151]: There is a `ExamNavigationBar` page object, which exposes the '#exam-navigation-bar' selector. I assume you want to hand in early here. The `ExamNavigationBar` already has a function for that

File: src/test/cypress/integration/exam/ExamAssessment.spec.ts [hunk 3]
```diff
@@ -124,11 +130,64 @@ describe('Exam assessment', () => {
...
+
+        it('Assesses quiz automatically', () => {
+            cy.waitUntil(() => {
+                const now = dayjs().isAfter(examEnd);
+                if (now) {
+                    return now;
+                } else {
+                    cy.wait(1000);
+                }
+            }).then(() => {
```
Comment (jhainzlschmid) [inline src/test/cypress/integration/exam/ExamAssessment.spec.ts:164]: `waitUntil` seems to be kind of an overkill.
Is the following not possible?
```typescript
const now = dayjs();
if (now.isBefore(examEnd)) {
    cy.wait(now.diff(examEnd));
}
```
Also the cypress commands are queued so chaining isn't needed here if I understand correctly?

File: src/test/cypress/integration/exam/ExamAssessment.spec.ts [hunk 4]
```diff
@@ -124,11 +130,64 @@ describe('Exam assessment', () => {
...
+                cy.contains('Evaluate quizzes').click();
+                cy.wait('@evaluateQuizzes').then(() => {
+                    cy.waitUntil(() => {
+                        const now = dayjs().isAfter(resultDate);
+                        if (now) {
+                            return now;
+                        } else {
+                            cy.wait(1000);
+                        }
+                    }).then(() => {
```
Comment (jhainzlschmid) [inline src/test/cypress/integration/exam/ExamAssessment.spec.ts:176]: Same question as above with the examEnd

**Review by jhainzlschmid — CHANGES_REQUESTED** [review]
Had some questions regarding the `waitUntil` usage and some suggestions for using existing page objects

**Review by jhainzlschmid — APPROVED** [review]
Suggestions have been implemented and related tests pass. Nice job 👍

**Pull request merged**